tests/utils.py

```diff
@@ -34,7 +58,7 @@ def run(
     callbacks.append(history)
     kwargs['callbacks'] = callbacks
 
-    with benchmarks(history):
+    with benchmarks(history, spec.work_dir):
```

Comment:
The `spec.work_dir` isn't the right thing to pass - it's not specific enough. I want the directory containing only the intermediate results written during the last cubed run.

I can see that each such directly is uniquely labeled which is nice, but I'm not sure exactly how to get this label from the computation. Should I import `cubed.core.plan.CONTEXT_ID` or is there a simpler way?

Reply:
The `event` object passed to `on_compute_start` (and `on_compute_end`) in the callback contains the `context_id` (see https://github.com/cubed-dev/cubed/blob/fda2f1e7bedec5389cae00ee751a54076ccf28e3/cubed/runtime/types.py#L28-L49). The `HistoryCallback` doesn't capture this - perhaps it could be changed, or maybe it's simpler to write a custom callback?

Reply:
In `cubed.core.plan` there is a `CONTEXT_ID` and a `compute_id`. The former seems to be used to make temporary directories, whilst the latter is what's stored in the `event` object. What's the difference between them?

Reply:
`CONTEXT_ID` is the directory name created for the duration of the client Python process. The `compute_id` is a recent addition used for creating directories to store history data in that is unique to the call to `compute`.

So you're right - you want `CONTEXT_ID`, but the intermediate data for all computations run in the pytest session will be there, so it's hard to pick out the arrays for a particular computation.

We should probably make Cubed store its intermediate data in a directory named `{CONTEXT_ID}/{compute_id}`, but that's a bit more work.

Reply:
Thanks for the explanation!

> We should probably make Cubed store its intermediate data in a directory named `{CONTEXT_ID}/{compute_id}`, but that's a bit more work.

I would like to do that - I can make a PR to cubed.

Reply:
(Continuing discussion from https://github.com/cubed-dev/cubed/issues/413)

> > I think the easier way to solve the original problem in https://github.com/cubed-dev/cubed-benchmarks/pull/10 would be to just get the intermediate array paths from the DAG
>
> So then the benchmark context managers need to know about the plan object right? Or can we add it to what's saved in `history.plan`?

I tried looking at the `plan` object but if this information is stored in there I can't seem to find it. I thought looking at `target` would give me what I need but for the smallest quad means example I just get `None` for each node, when clearly something was written.

I'm also wondering if there is any subtlety here with the optimized vs unoptimized plan?

Reply:
The plan only has the op names now. Optimized vs unoptimized shouldn't make a difference. Try something like (untested)

```python
array_names = [name for name in dag if name.startswith("array-")]
```

Where `dag` comes from the callback object.